relational_operators/WorkOrderFactory.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ *   Copyright 2015-2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_RELATIONAL_OPERATORS_WORK_ORDER_FACTORY_HPP_
+#define QUICKSTEP_RELATIONAL_OPERATORS_WORK_ORDER_FACTORY_HPP_
+
+#include "utility/Macros.hpp"
+
+#include "tmb/id_typedefs.h"
+
+namespace tmb { class MessageBus; }
+
+namespace quickstep {
+
+class CatalogDatabaseLite;
+class QueryContext;
+class StorageManager;
+class WorkOrder;
+
+namespace serialization { class WorkOrder; }
+
+/** \addtogroup RelationalOperators
+ *  @{
+ */
+
+/**
+ * @brief All-static factory object that provides access to WorkOrder.
+ **/
+class WorkOrderFactory {
+ public:
+  /**
+   * @brief Get a pointer to a WorkOrder from its serialized Protocol Buffer
+   *        form.
+   *
+   * @param proto The Protocol Buffer representation of a WorkOrder object,
+   *        originally generated by RelationalOperator::getAllWorkOrders.
+   * @param catalog_database The database to resolve relation and attribute
+   *        references in.
+   * @param query_context A pointer to QueryContext.
+   * @param storage_manager The StorageManager to use.
+   * @param shiftboss_client_id The TMB client id of Shiftboss.
+   * @param bus A pointer to the TMB.
+   *
+   * @return A new WorkOrder reconstructed from the supplied Protocol Buffer.
+   **/
+  static WorkOrder* ReconstructFromProto(const serialization::WorkOrder &proto,
```

Comment:
I have some questions: 
1. Will this function be called by ShiftBoss?
2. Will this function be called once per WorkOrder? 
3. Are there any opportunities for serializing multiple WorkOrders in one proto and subsequently deserializing them all at once? It should be possible for the Foreman to dispatch multiple WorkOrders to one Shiftboss at once. Therefore, we should be able to gain some saving in bulk serializing and deserializing of WorkOrders.